src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -34,31 +41,74 @@
  * name but different signatures where the signature can differ by the number of
  * input parameters or type of input parameters or both.
  * </p>
+ * <ul>
+ * <li>
+ * Property {@code modifierGroups} - control how different types of overloaded methods
+ * can be grouped by their modifiers.
+ * Type is {@code java.util.regex.Pattern[]}.
+ * Default value is {@code ""}.
+ * </li>
+ * </ul>
  * <p>
  * To configure the check:
  * </p>
  * <pre>
- * &lt;module name=&quot;OverloadMethodsDeclarationOrder&quot;/&gt;
+ * &lt;module name="OverloadMethodsDeclarationOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example of incorrect grouping of overloaded methods:
+ * </p>
+ * <pre>
+ * public void foo(int i) {} // OK
+ * public void foo(String s) {} // OK
+ * public void notFoo() {} // OK
+ * public void foo(String s, int i) {} // Violation, should be after 'foo(int i, String s)'
  * </pre>
  * <p>
  * Example of correct grouping of overloaded methods:
  * </p>
  * <pre>
- * public void foo(int i) {}
- * public void foo(String s) {}
- * public void foo(String s, int i) {}
- * public void foo(int i, String s) {}
- * public void notFoo() {}
+ * public void foo(int i) {} // OK
+ * public void foo(String s) {} // OK
+ * public void foo(String s, int i) {} // OK
+ * public void foo(int i, String s) {} // OK
+ * public void notFoo() {} // OK
  * </pre>
  * <p>
- * Example of incorrect grouping of overloaded methods:
+ * To configure the check to group static and private methods separately within their own
+ * individual groups
+ * </p>
+ * <pre>
+ * &lt;module name="OverloadMethodsDeclarationOrder"&gt;
+ *   &lt;property name="modifierGroups" value="static, private"&gt;&lt;/property&gt;
```

Comment:
In this example public isn't a modifier set in the options.

Does this mean for `public` methods they can be in any order compared to private/static? What about if a public is between 2 private/static, can it split the 2 even though its not defined? We need some verbage and example showing what it means when a modifier does not show here.

If it makes sense, it could be a notes.
https://checkstyle.org/config_blocks.html#EmptyCatchBlock_Notes

Reply:
Yes, the modifier groups only specify what methods must be grouped together, where they appear in relation to each other does not matter as long as they are grouped correctly. In this example, public must appear next to any other methods that are neither static nor private.